### PR TITLE
Make application plans publication state configurable

### PR DESCRIPTION
**Pull request by miguelsorianod**
Implement Applications Plan publication configurability in a Product.
A `published` attribute has been implemented to control it, and the Product CR looks like this:

```
...
applicationPlans:
 production:
   name: "production"
   published: true
...
```

If not specified, by default it is `false`.

Pending:

- [x] Verification

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Ready for review @eguzki

File: controllers/capabilities/application_plan_reconciler.go [hunk 1]
```diff
@@ -111,6 +111,18 @@ func (a *applicationPlanReconciler) syncPlan(_ interface{}) error {
 		}
 	}
 
+	planEntityStateIsPublished := a.planEntity.State() == "published" // If the state is not published then we assume it is "hidden"
+	desiredApplicationPlanIsPublished := a.resource.Published != nil && *a.resource.Published
```
Comment (eguzki) [inline controllers/capabilities/application_plan_reconciler.go:115]: minor thing: better to implement `Published()` method in the `capabilitiesv1beta1.ApplicationPlanSpec`

File: apis/capabilities/v1beta1/product_types.go [hunk 2]
```diff
@@ -162,6 +162,11 @@ type ApplicationPlanSpec struct {
 	// +optional
 	Limits []LimitSpec `json:"limits,omitempty"`
 
+	// Controls whether the application plan is published. If not specified it is
+	// hidden by default
+	// optional
```
Comment (eguzki) [inline apis/capabilities/v1beta1/product_types.go:167]: :eyes:

**Review by eguzki — CHANGES_REQUESTED** [review]
minor things. Looking good.

File: apis/capabilities/v1beta1/product_types.go (hunk 2 above)
Comment (miguelsorianod) [reply apis/capabilities/v1beta1/product_types.go]: :+1:  fixed

File: controllers/capabilities/application_plan_reconciler.go (hunk 1 above)
Comment (miguelsorianod) [reply controllers/capabilities/application_plan_reconciler.go:115]: :+1: created

Comment (eguzki) [reply controllers/capabilities/application_plan_reconciler.go:115]: I do not see it

Comment (miguelsorianod) [reply controllers/capabilities/application_plan_reconciler.go]: you should see it now

File: controllers/capabilities/application_plan_reconciler.go [hunk 3]
```diff
@@ -111,6 +111,18 @@ func (a *applicationPlanReconciler) syncPlan(_ interface{}) error {
 		}
 	}
 
+	planEntityStateIsPublished := a.planEntity.State() == "published" // If the state is not published then we assume it is "hidden"
+	desiredApplicationPlanIsPublished := a.resource.IsPublished()
```
Comment (eguzki) [inline controllers/capabilities/application_plan_reconciler.go:115]: no need to keep it in a separate variable, but good enough

**Review by eguzki — APPROVED** [review]

**Comment by miguelsorianod** [comment]
added documentation in product reference doc

**Pull request merged**